### PR TITLE
documentation for opensearch multi env configuration

### PR DESCRIPTION
**Pull request by lazpavel**
_Description of changes:_
- added documentation for configuring different open search instance types in a multi environment setup

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/cli/graphql-transformer/searchable.md [hunk 1]
```diff
@@ -130,6 +130,40 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
 - Support for adding the `@searchable` directive does not yet provide automatic indexing for any existing data to OpenSearch. View the feature request [here](https://github.com/aws-amplify/amplify-cli/issues/98).
+- `T2.small` OpenSearch instance type is not recommended to be used in a production environment.
+
+### Configure environment OpenSearch instance type
+
+By default Amplify CLI will configure a `T2.small` instance type. This is not recommended to be used in the production environment as per the OpenSearch [best practices](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/bp.html).
+You can configure the OpenSearch instance type per environment as follows.
+
+1. Run `amplify env add` and set up an environment (e.g. "prod")
+2. Edit the `amplify/team-provider-info.json` file and set `ElasticSearchInstanceType` to the instance type that works for your application
```
Comment (offlineprogrammer) [inline docs/cli/graphql-transformer/searchable.md:141]: We need to rename all instances of Elasticsearch in our docs, libraries, tools and marketing pages. The new name is “OpenSearch”

**Review by offlineprogrammer — COMMENTED** [review]
We need to rename all instances of Elasticsearch in our docs, libraries, tools and marketing pages. The new name is “OpenSearch”

File: docs/cli/graphql-transformer/searchable.md (hunk 1 above)
Comment (lazpavel) [reply docs/cli/graphql-transformer/searchable.md]: I used `OpenSearch` term for all free text, however we still use `ElasticSearch...` in many of our enum values, the docs for those will need to change once we have the changes in our libraries

File: src/pages/cli/graphql-transformer/searchable.mdx [hunk 2]
```diff
@@ -130,6 +130,40 @@ Here is a complete list of searchable operations per GraphQL type supported as o
 - `@searchable` is not compatible with Amazon ElasticSearch t2.micro instance as it only works with ElasticSearch version 1.5 and 2.3 and Amplify CLI only supports instances with ElasticSearch version >= 6.x.
 - `@searchable` is not compatible with the @connection directive
 - Support for adding the `@searchable` directive does not yet provide automatic indexing for any existing data to OpenSearch. View the feature request [here](https://github.com/aws-amplify/amplify-cli/issues/98).
+- `T2.small` OpenSearch instance type is not recommended to be used in a production environment.
+
+### Configure environment OpenSearch instance type
+
+By default Amplify CLI will configure a `T2.small` instance type. This is not recommended to be used in the production environment as per the OpenSearch [best practices](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/bp.html).
+You can configure the OpenSearch instance type per environment as follows.
```
Comment (attilah) [inline src/pages/cli/graphql-transformer/searchable.mdx:138]: as `follows:` ?

File: src/pages/cli/graphql-transformer/searchable.mdx [hunk 3]
```diff
@@ -130,6 +130,40 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+
+1. Run `amplify env add` and set up an environment (e.g. "prod")
```
Comment (attilah) [inline src/pages/cli/graphql-transformer/searchable.mdx:140]: "to create a new environment (e.g. "prod")" ?

File: src/pages/cli/graphql-transformer/searchable.mdx [hunk 4]
```diff
@@ -130,6 +130,40 @@ Here is a complete list of searchable operations per GraphQL type supported as o
 - `@searchable` is not compatible with Amazon ElasticSearch t2.micro instance as it only works with ElasticSearch version 1.5 and 2.3 and Amplify CLI only supports instances with ElasticSearch version >= 6.x.
 - `@searchable` is not compatible with the @connection directive
 - Support for adding the `@searchable` directive does not yet provide automatic indexing for any existing data to OpenSearch. View the feature request [here](https://github.com/aws-amplify/amplify-cli/issues/98).
+- `T2.small` OpenSearch instance type is not recommended to be used in a production environment.
```
Comment (renebrandel) [inline src/pages/cli/graphql-transformer/searchable.mdx:133]: isn't it lower case "t2"?

File: src/pages/cli/graphql-transformer/searchable.mdx [hunk 5]
```diff
@@ -130,6 +130,40 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+
+### Configure environment OpenSearch instance type
+
+By default Amplify CLI will configure a `T2.small` instance type. This is not recommended to be used in the production environment as per the OpenSearch [best practices](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/bp.html).
```
Comment (renebrandel) [inline src/pages/cli/graphql-transformer/searchable.mdx:137]: Same question as above, shouldn't it be lower case "t2"?

Comment (renebrandel) [inline src/pages/cli/graphql-transformer/searchable.mdx:137]:
```suggestion
By default Amplify CLI will configure a `T2.small` instance type. This is great for getting started and prototyping BUT not recommended to be used in the production environment per the OpenSearch [best practices](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/bp.html).
```

File: src/pages/cli/graphql-transformer/searchable.mdx [hunk 6]
```diff
@@ -130,6 +130,40 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+            "ElasticSearchInstanceType": "t2.medium.elasticsearch"
+          }
+        }
+      }
+    }
+  }
+```
+3. Deploy your changes with `amplify push`
+
+You can find more information about Amazon OpenSearch Service supported instance types [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-instance-types.html).
```
Comment (renebrandel) [inline src/pages/cli/graphql-transformer/searchable.mdx:166]:
```suggestion
Learn more about Amazon OpenSearch Service instance types [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-instance-types.html).
```

**Review by renebrandel — CHANGES_REQUESTED** [review]

File: src/pages/cli/graphql-transformer/searchable.mdx (hunk 4 above)
Comment (lazpavel) [reply src/pages/cli/graphql-transformer/searchable.mdx]: updated

**Review by renebrandel — APPROVED** [review]

**Review by kaustavghosh06 — APPROVED** [review]

**Pull request merged**